src/riak_cs_stats.erl

```diff
@@ -227,88 +133,94 @@ report_pretty_json() ->
      {list_to_atom(lists:flatten([Name, $_, "overflow"])), PoolOverflow},
      {list_to_atom(lists:flatten([Name, $_, "size"])), PoolSize}].
 
-
+metric_to_atom(Key, Suffix) ->
+    StringKey = string:join([atom_to_list(Token) || Token <- Key], "_"),
+    list_to_atom(lists:flatten([StringKey, Suffix])).
 
 -ifdef(TEST).
 
 -include_lib("eunit/include/eunit.hrl").
 
-stats_metric_test() ->
-    [begin
-         ?debugVal(Key),
-         case lists:last(Key) of
-             time ->
-                 ?assertEqual(histogram, Type),
-                 [?assert(proplists:is_defined(M, Aliases))
-                  || M <- [mean, median, 95, 99, 100]];
-             _ ->
-                 ?assertNotEqual(false, lists:keyfind(Key, 1, ?METRICS)),
-                 ?assertEqual(spiral, Type),
-                 ?assert(proplists:is_defined(one, Aliases)),
-                 ?assert(proplists:is_defined(count, Aliases))
-         end,
-         ?assertEqual([], Options)
-     end || {Key, Type, Options, Aliases} <- ?METRICS].
-
 stats_test_() ->
     Apps = [setup, compiler, syntax_tools, goldrush, lager, exometer_core],
     {setup,
      fun() ->
              [ok = application:start(App) || App <- Apps],
-             ok = riak_cs_stats:init()
+             ok = init()
      end,
      fun(_) ->
              [ok = application:stop(App) || App <- Apps]
      end,
      [{inparallel, [fun() ->
-                            %% ?debugVal(Key),
-                            case lists:last(Key) of
-                                time -> ok;
-                                _ -> riak_cs_stats:update(Key, 16#deadbeef)
-                            end
-                    end || {Key, _, _, _} <- ?METRICS]},
-     fun() ->
-             [begin
-                  Items = raw_report_item(I),
-                  %% ?debugVal(Items),
-                  case length(Items) of
-                      2 ->
-                          ?assertEqual([1, 1],
-                                       [N || {_, N} <- Items]);
-                      5 ->
-                          ?assertEqual([16#deadbeef, 16#deadbeef, 16#deadbeef, 16#deadbeef, 0],
-                                       [N || {_, N} <- Items])
-                  end
-              end || I <- ?METRICS]
-     end]}.
+                            inflow(Key),
+                            update(Key, 16#deadbeef),
+                            update([error | Key], 16#deadbeef)
+                    end || Key <- duration_metrics()]},
+      fun() ->
+              [begin
+                   Report = [N || {_, N} <- report_duration(Key, SubKey, ExometerType)],
+                   case ExometerType of
+                       spiral ->
+                           ?assertEqual([1, 1], Report);
+                       histogram ->
+                           ?assertEqual(
+                              [16#deadbeef, 16#deadbeef, 16#deadbeef, 16#deadbeef, 0],
```

Comment:
This code is actually written by me, but isn't it strange that `some_metric_100` returns `0` even if we update the value. This ls smaller than `some_metric_99` or others.